src/app/host/[room]/page.tsx

```diff
@@ -1,46 +1,150 @@
 // HOST ROOM
 'use client'
 
+import Link from 'next/link'
 import { useEffect, useState } from 'react'
-import SocketIoInfo from '@/components/socketIoDevTools/SocketIoInfo'
-
 import { socketRoomEmitter } from '@/services/socket'
+import { useSocketListener } from '@/services/socket'
+import type { ListenerRes } from '@/services/socket'
+import HostControlButton from '@/components/HostControlButton'
 import HostSettingsButton from '@/components/HostSettingsButton'
-import HostControls from '@/components/HostControls'
 import RoomMainUi from '@/components/RoomMainUi'
+import RoomInfo from '@/components/RoomInfo'
 
-export default function HostRooms({ params }: { params: { room: string } }) {
-	const { room } = params
+export default function HostRoom({ params }: { params: { room: string } }) {
+	const [allUserPointData, setAllUserPointData] = useState<ListenerRes[]>([])
+	const storyPointRes = useSocketListener('story-points')
+	const joinRoomRes = useSocketListener('join-room')
 
-	console.log('%c>>> HOST ROOM roomId:', 'color: #5f0', room)
+	useEffect(() => {
+		if (!storyPointRes) return
+
+		console.log('%c>>> listener', 'color: red', storyPointRes)
+		setAllUserPointData((prevUsersPoints) => {
+			const index = prevUsersPoints.findIndex((data) => {
+				return data.userName === storyPointRes.userName
+			})
+			if (index !== -1) {
+				const noDuplicates = [...prevUsersPoints]
+				noDuplicates[index].message = storyPointRes.message
+				console.log('%c>>> noDuplicates SP', 'color: red', noDuplicates)
+				return noDuplicates
+			}
+			return [...prevUsersPoints, storyPointRes]
+		})
+	}, [storyPointRes])
+
+	useEffect(() => {
```

Comment:
todo: This 2 `useEffect` can be replaced by the socket callback pattern we spoke in our call :)

Reply:
Same as above.